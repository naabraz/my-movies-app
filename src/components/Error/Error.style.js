import styled from 'styled-components/native';
import LottieView from 'lottie-react-native';

const Container = styled.View`
  flex: 1;
  justify-content: center;
  align-self: center;
`;

const Animation = styled(LottieView)`
  width: 200px;
`;

export { Container, Animation };
