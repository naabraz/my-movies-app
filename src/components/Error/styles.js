import styled from 'styled-components/native';
import LottieView from 'lottie-react-native';

export const Container = styled.View`
  flex: 1;
  justify-content: center;
  align-self: center;
`;

export const Animation = styled(LottieView)`
  width: 200px;
`;
