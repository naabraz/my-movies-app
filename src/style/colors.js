export const brightGray = '#393e46';
export const ebonyClay = '#222831';
export const bondiBlue = '#00adb5';
export const gallery = '#eeeeee';
