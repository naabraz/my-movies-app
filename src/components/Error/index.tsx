import React from 'react';

import { Container, Animation } from './styles';

const Error: React.FC = () => (
  <Container>
    <Animation
      source={require('assets/lottie/error.json')}
      autoPlay
      loop
      hardwareAccelerationAndroid
      testID="ErrorAnimation"
    />
  </Container>
);

export default Error;
