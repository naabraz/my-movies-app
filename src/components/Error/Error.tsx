import React from 'react';

import * as Styled from './Error.style';

export const Error: React.FC = () => (
  <Styled.Container>
    <Styled.Animation
      source={require('../../../assets/lottie/error.json')}
      autoPlay
      loop
      hardwareAccelerationAndroid
    />
  </Styled.Container>
);
