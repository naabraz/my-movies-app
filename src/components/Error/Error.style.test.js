import 'jest-styled-components/native';
import React from 'react';
import renderer from 'react-test-renderer';

import * as Styled from './Error.style';

describe('Given Error component style', () => {
  it('Should render Container component correctly', () => {
    const errorComponent = renderer.create(<Styled.Container />).toJSON();
    expect(errorComponent).toHaveStyleRule('justify-content', 'center');
    expect(errorComponent).toHaveStyleRule('align-self', 'center');
  });
});
