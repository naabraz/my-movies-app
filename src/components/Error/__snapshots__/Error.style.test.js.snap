// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Error component style Should render Animation component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "aspectRatio": NaN,
      },
      undefined,
      Array [
        Object {
          "width": 200,
        },
      ],
    ]
  }
>
  <View
    autoPlay={false}
    enableMergePathsAndroidForKitKatAndAbove={false}
    loop={true}
    onAnimationFinish={[Function]}
    progress={0}
    resizeMode="contain"
    sourceJson="{}"
    speed={1}
    style={
      Object {
        "aspectRatio": NaN,
        "height": "100%",
        "width": 200,
      }
    }
  />
</View>
`;

exports[`Given Error component style Should render Container component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
/>
`;
